big_tests/tests/s2s_SUITE.erl

```diff
@@ -114,9 +115,30 @@ init_per_group(GroupName, Config) ->
 end_per_group(_GroupName, _Config) ->
     ok.
 
+init_per_testcase(dns_discovery = CaseName, Config) ->
+    ok = rpc(mim(), meck, new, [inet_res, [no_link, unstick, passthrough]]),
+    ok = rpc(mim(), meck, expect, [inet_res, getbyname, 3,
+             {ok, {hostent, "_xmpp-server._tcp.fed2", [], srv, 1, [{30, 10, 5299, "localhost"}]}}]),
+
+    ok = rpc(mim(), meck, new, [inet, [no_link, unstick, passthrough]]),
+    ok = rpc(mim(), meck, expect, [inet, getaddr, 2, {ok, {127, 0, 0, 1}}]),
```

Comment:
I don't know where else in code are `inet` and `inet_res` going to be called, so these mocks seem dangerous to me. Can you try to mock only on the specific input in question and passthrough for all other?

Also, I believe the 4-tuple in the address list should contain ip_tuples, not domains, so `{127, 0, 0, 1}` instead of `"localhost"`, see the return type of `do_lookup_services/2`.

Reply:
Thanks for pointing that out, I've changed the mock.

I checked the documentation, and `inet_res:get_by_name/3` returns `{Prio :: integer(), Weight :: integer(), Port :: integer(), dns_name()}` in the address list, where `dns_name()` is just a string. I also tested it manually with a real DNS server, and it also returned a string. I think there was an issue with the typespec for `do_lookup_services/2`